src/tests/topp/IDFileConverter_9_output.mzid

```diff
@@ -203,7 +203,7 @@
 					<cvParam accession="MS:1001143" cvRef="PSI-MS" name="PSM-level search engine specific statistic"/>
 					<userParam name="MOWSE" unitName="xsd:double" value="1.4"/>
 				</SpectrumIdentificationItem>
-				<cvParam accession="MS:1000894" cvRef="PSI-MS" name="retention time" value="1234.5" unitAccession="UO:0000010" unitName="second" unitCvRef="UO"/>
+				<cvParam accession="MS:1000894" cvRef="PSI-MS" name="retention time" value="1234.5" unitAccession="UO:0000010" unitCvRef="UO"/>
```

Comment:
here we dont have a solution (yet)

Reply:
I used a slightly bigger consensusXML with more ID data.  
Before flat_map: 3297 MB
After flat_map: 2334 MB 
After slim data value: 1952 MB. 
I think this is a nice improvement. Also FileInfo is done from 3:00m to 2:40m

Reply:
with #pragma pack(push, 1) 
and #pragma pack(pop)
I get: 1857 MB but 2:52m

Reply:
> with #pragma pack(push, 1)

great for checking, I wonder though whether that is a good idea for a library? I am a bit worried whether that may lead to binary incompatibilities

>  I get: 1857 MB but 2:52m

so it actually takes longer as expected due to mis-alignment, however I dont understand how you can gain another 95 MB because it seems like you are almost saving another 10 bytes here - but how would that be possible on a struct of 16 bytes that only has 1 byte of padding?

Can you check the `sizeof(DataValue)` before and after the pragma?

Reply:
> After flat_map: 2334 MB
> After slim data value: 1952 MB.
> After pack:  1857 MB but 2:52m

so I am wondering how you go from 2334 -> 1952 (382 MB) when going from 56 to 16 bytes but then you gain another 95 MB going from 16 to 14 / 12 where the math would predict that a gain of 95 MB should be another gain of 8 bytes.  Maybe it has something to do with how `vector` stores packed structs?

Reply:
> very cool. thanks! I will try that branch on the data and report back. So I guess the lookup is still missing in that part?

yes the lookup is missing and I am not sure that its very important.